### PR TITLE
Refactor CEBtoCEPs and CEPtoCEB maps

### PR DESCRIPTION
**Pull request by krishgobinath**
1) Use single lock to protect both the CEBtoCEPs and CEPtoCEB
maps.

Currently we have 2 maps, 

1. `desiredCebMapping`  which is used to store desiredCEB state in local cache and maintains a map of `CEB name` to list of `CiliumEndpoints`, this map is protected by a separate mutex for thread safe access.
2. `cacheCEP` is inverse of `desiredCebMapping`, which maps `CEP name` to `CEB name`, and this map is protected by a separate mutex for thread safe access.

These 2 maps stores same information, this need to be protected by single mutex to avoid potential race condition issues.

This PR refactors these map implementation, uses single map to protect data for safe access.

Signed-off-by: Gobinath Krishnamoorthy <gobinathk@google.com>

### CODE REVIEW

**Review by Weil0ng — APPROVED** [review]
Change LGTM, can you please update the commit msg/PR description to elaborate what was the issue and how this fixes it?

File: operator/pkg/ciliumendpointbatch/ceptocebmap.go [hunk 1]
```diff
@@ -19,121 +19,110 @@ import "github.com/cilium/cilium/pkg/lock"
...
+// These maps are used by the CEB manager, primarily used for storing and retrieving
+// the desired CEBs by thread-safe.
 // This map is protected by lock for consistent and concurrent access.
-type cepToCebMapping struct {
-	cepMutex lock.RWMutex
-	cacheCep map[CEPName]CEBName
+type CEBToCEPMapping struct {
+	cebMutex lock.RWMutex
+	// CEPNameToCEBName is used to map CiliumEndpoint name to CiliumEndpointBatch name.
+	CEPNameToCEBName map[CEPName]CEBName
```
Comment (aanm) [inline operator/pkg/ciliumendpointbatch/ceptocebmap.go]: this field should be made private at the risk of being accessed outside this package without holding the RWMutex.

**Review by aanm — CHANGES_REQUESTED** [review]
Please add the PR description in the commit message as well. Thank you

File: operator/pkg/ciliumendpointbatch/ceptocebmap.go (hunk 1 above)
Comment (krishgobinath) [reply operator/pkg/ciliumendpointbatch/ceptocebmap.go]: Good point.

**Review by aanm — APPROVED** [review]

**Pull request merged**